dashboard/test/models/unit_deletion_script_test.rb

```diff
@@ -0,0 +1,46 @@
+require 'test_helper'
+require 'cdo/shared_constants'
+
+# Includes assertions on the Unit model that the unit deletion script
+# takes a dependency on.
+class UnitDeletionScriptTest < ActiveSupport::TestCase
+  include Curriculum::SharedCourseConstants
+  self.use_transactional_test_case = true
+
+  setup_all do
+  end
+
+  setup do
+    UnitGroup.clear_cache
+    Unit.clear_cache
+  end
+
+  test 'get published state for stand alone' do
+    unit = create(:script, published_state: PUBLISHED_STATE.in_development)
+    assert_equal PUBLISHED_STATE.in_development, unit.get_published_state
+  end
+
+  test 'get published state for unit in unit group with nil published state' do
+    unit = create(:script, published_state: nil)
+    unit_gp = create(:unit_group, published_state: PUBLISHED_STATE.in_development)
+    create :unit_group_unit, unit_group: unit_gp, script: unit, position: 1
+
+    assert_equal PUBLISHED_STATE.in_development, unit.get_published_state
+  end
+
+  test 'get published state unit in_development and unit group stable' do
+    unit = create(:script, published_state: PUBLISHED_STATE.in_development)
+    unit_gp = create(:unit_group, published_state: PUBLISHED_STATE.stable)
+    create :unit_group_unit, unit_group: unit_gp, script: unit, position: 1
+
+    assert_equal PUBLISHED_STATE.in_development, unit.get_published_state
+  end
+
+  test 'get published state unit stable and unit group in_development' do
+    unit = create(:script, published_state: PUBLISHED_STATE.stable)
+    unit_gp = create(:unit_group, published_state: PUBLISHED_STATE.in_development)
+    create :unit_group_unit, unit_group: unit_gp, script: unit, position: 1
+
+    assert_equal PUBLISHED_STATE.stable, unit.get_published_state
+  end
+end
```

Comment:
I think I see what you're doing here, but this approach makes it difficult to see how these tests provide coverage for what's going on in bin/curriculum/delete_unit.rb . my recommendation would be to extract the parts you want to cover with unit tests into dashboard/app/helpers/delete_unit_helper.rb, and then have the unit test execute that code directly. once you've done that, this test file should also move to dashboard/test/helpers.

Reply:
I hesitate to move the unit deletion code to dashboard yet as its written with assumptions around using in the script and following through instructions from a doc. Moving to dashboard might open it up for other unintended usage.

Regarding your question on how these unit tests help, the idea was more like integration tests, where we can use these unit tests to assert the behaviors that the deletion script is assuming at this time. If any of the published state logic were to change in a way that would break the assumption, this could catch that.

Reply:
Can we discuss this live some time? I'm not sure I agree with the practice of duplicating the code under test just because we are worried about someone finding and using the code, and I'm having trouble imagining the scenario where someone would find and use the code incorrectly just because it is in the dashboard directory. My concern with the current approach is that if the bin/curriculum script changes, there's no chance of catching an error with this test.

if dashboard/app/helpers seems problematic as a location, one alternative is to put this in dashboard/lib/services.

Reply:
As per our discussion, I'll move the logic around "IsSafeToDelete" to the lib directory, potentially under policies and include these UTs, along with other tests to validate cases around sections and progress there.